### PR TITLE
added 'Clarinet'

### PR DESCRIPTION
**Pull request by corno**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] Add or edit tests to reflect the change. (Run with `npm test`.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [X] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [X] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [X] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [X] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` should be present and it shouldn't have any additional or disabling of rules. Just content as `{ "extends": "dtslint/dt.json" }`. If for reason the some rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]`  and not for whole package so that the need for disabling can be reviewed.
- [X] `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/clarinet/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,98 @@
+// Type definitions for clarinet 0.12
+// Project: https://github.com/dscape/clarinet
+// Definitions by: corno <https://github.com/corno>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+// import { Stream } from "stream";
```
Comment (jessetrinity) [inline types/clarinet/index.d.ts]: Did you intend to remove?

File: types/clarinet/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    onerror: (e: Error) => void;
+    onready: () => void;
+    constructor(opt?: Options);
+    end(): void;
+    write(chunk: string): void;
+    resume(): void;
+    close(): void;
+}
+
+export class CStream /*extends Stream*/ {
```
Comment (jessetrinity) [inline types/clarinet/index.d.ts]: this one too?

File: types/clarinet/index.d.ts (hunk 1 above)
Comment (corno) [reply types/clarinet/index.d.ts]: I do not know how to import from the @types/node. Can you tell me how to?

Comment (jessetrinity) [reply types/clarinet/index.d.ts]: `///<reference types="node">`

Read the entry on [this page](https://www.typescriptlang.org/docs/handbook/triple-slash-directives.html#-reference-types-) to verify whether that is what you're looking for.

Comment (jessetrinity) [reply types/clarinet/index.d.ts]: The imports I commented on are commented out though, so it looks like they were left in on accident?

Comment (corno) [reply types/clarinet/index.d.ts]: well, they should have been active, but I did not know how to, so thank you for the pointer.

If I add ````///<reference types="node">```` It still reports that it cannot find module "stream"

Comment (jessetrinity) [reply types/clarinet/index.d.ts]: Sorry :(
```ts
/// <reference types="node"/>
import { Stream } from "stream";
```
I left out the final `/`

Comment (corno) [reply types/clarinet/index.d.ts]: Thank you. That did it. I'll create a pull request right now.

**Comment by corno** [comment]
@jessetrinity The pr is ready. Let me know if it is okay like this.

**Comment by jessetrinity** [comment]
Awesome. Thanks for contributing!

**Comment by corno** [comment]
and thank you for your help!

**Pull request merged**